### ISSUE TITLE
NX@11 support

### ISSUE THREAD

**Issue by virus2016** [bennymeg/nx-electron#76]
**Describe the bug**
I have an NX@11 workspace and trying to add nx electron.

**To Reproduce**
```
yarn nx g nx-electron:app desktop --frontendProject=web                
yarn run v1.22.10
$ nx g nx-electron:app desktop --frontendProject=web
? In which directory should the electron application be generated? 
✔ Packages installed successfully.
CREATE apps/desktop/.eslintrc.json
CREATE apps/desktop/tsconfig.app.json
CREATE apps/desktop/tsconfig.json
CREATE apps/desktop/src/main.ts
CREATE apps/desktop/src/app/app.ts
CREATE apps/desktop/src/app/constants.ts
CREATE apps/desktop/src/app/api/preload.ts
CREATE apps/desktop/src/app/events/electron.events.ts
CREATE apps/desktop/src/app/events/squirrel.events.ts
CREATE apps/desktop/src/app/events/update.events.ts
CREATE apps/desktop/src/app/options/maker.options.json
CREATE apps/desktop/src/assets/.gitkeep
CREATE apps/desktop/src/environments/environment.prod.ts
CREATE apps/desktop/src/environments/environment.ts
CREATE apps/desktop/jest.config.js
CREATE apps/desktop/tsconfig.spec.json
UPDATE package.json
UPDATE .vscode/extensions.json
UPDATE workspace.json
UPDATE nx.json
UPDATE jest.config.js
✨  Done in 23.38s.
/apps (nxElectron) > nx serve desktop

> nx run desktop:serve 
Invalid format version detected - Expected:[ 1 ] Found: [ 2 ]
/apps (nxElectron) > 
```

**Expected behavior**
It should start electron.

**Desktop (please complete the following information):**
 - OS: MACOS
 - Nx Electron Version 11

**Additional context**
Workspace verison is 2.

**Comment by bennymeg** [comment]
Hi,
As described on #66, the _alpha_ version currently only support nx workspace v1 or angular workspace (angular.json / workspace.json@v1).

I will slowly migrate @angular-devkit to @nrwl/devkit in order to resolve this issue (as explained by __pascalbayer__ [here](https://github.com/bennymeg/nx-electron/issues/66#issuecomment-775717802)).

**Comment by s4m0r4m4** [comment]
@bennymeg  let me know if there's anything specific you'd like help with in upgrading to Angular and Nx 11.

**Comment by virus2016** [comment]
Is there anything we can do to help?

**Comment by bennymeg** [comment]
There are three features to develop:
- add workspace v2 support
- add [generatePackageJson](https://github.com/nrwl/nx/blob/master/packages/node/src/utils/generate-package-json.ts) support
- fix all test cases

The most important feature is to add support to workspace v2, but I think I managed to implement it.
I think fixing the test cases is quite a big task, so if you would like to help adding generatePackageJson support it would be nice.
Also, if you could verify that `alpha.2` works properly for you it would be wonderful.

**Comment by virus2016** [comment]
Thanks @bennymeg! I will give it a go today and see how it works. 

I could give the generatePackageJson support a try but I am not sure where it should be called? 

Not sure if I am the best person to help now 😂.

**Comment by virus2016** [comment]
It's working - Great job @bennymeg!

Couple of observations:

- Can't just run serve: You need to run serve with your "web" app before starting nx-electron. Not sure if this is default behaviour
- Package doesn't contain the JS output files in dist. It has no js files that I can see. 

I guess that some if not all will be fixed by the generatePackageJson.

**Comment by bennymeg** [comment]
> It's working - Great job @**bennymeg**!

Great!

> I could give the generatePackageJson support a try but I am not sure where it should be called?
> Not sure if I am the best person to help now 😂.

Dont worry about it then, I'll take care of it later.

> Can't just run serve: You need to run serve with your "web" app before starting nx-electron.

That is the expected behavior, that's the reason you have hot reloading for your frontend and backend projects.

> Package doesn't contain the JS output files in dist. It has no js files that I can see.

You need to build both frontend and backend projects before you package / make them.

**Comment by virus2016** [comment]
> You need to build both frontend and backend projects before you package / make them.

Yeah, did that however the package isn't copying the JS files over... 

Here is my dist folder: 
![image](https://user-images.githubusercontent.com/11227384/108869841-7e357c00-75ef-11eb-9aa3-74b14cd1a7aa.png)

![image](https://user-images.githubusercontent.com/11227384/108870981-93f77100-75f0-11eb-827f-429456195b54.png)

It could just be dir/path issue?

**Comment by virus2016** [comment]
Does the js files get put into the electron package folder under resources? 

![image](https://user-images.githubusercontent.com/11227384/108871281-e769bf00-75f0-11eb-8757-e4ebf31a1a22.png)

**Comment by virus2016** [comment]
It was my fault! Forgot basehref!

**Comment by s4m0r4m4** [comment]
I'm not familiar with the generatePackegJson stuff, I'll take a look at contributing some test fixes

**Comment by s4m0r4m4** [comment]
@bennymeg - trying to get tests to run, but I'm a bit of a noob when it comes to jest testing (call me old-fashioned, I just use karma haha). 

`npm run test` yields the following error: `No tests found, exiting with code 1`, which makes sense because all the test files are *spec.**ts** extension, while the npm script explicitly tells jest to look for *spec.**js** files. I tried running `npm run compile` to produce the javascript output, but now hit the following error:
`Path "C:\<my-path-here>\nx-electron\build\collection.json" does not exist.`

**Comment by bennymeg** [comment]
Thanks,
I already took care of generatePackegJson here (281ace08dc06251fb6c9a4f8717dcc66b4c81ba0).

The reason I use the spec.js file is that jest can't test typescript natively (probebly a good time to add `ts-jest`, I'll take care of it). 
For now you can try running `npm rub build`, this will also copy all the relevant files to the build directory.

**Comment by s4m0r4m4** [comment]
Ok, I tried running the tests and was able to get them to work with a few tweaks. 

I'm seeing a ton of broken tests from trying to spy on exported functions from the devkitArchitect that are not declared configurable. Were these working before the v11 bump? Based on my research it seems they've likely been broken for a long time.

**Comment by bennymeg** [comment]
@s4m0r4m4 Thanks for helping.
Some of the tests was broken for a long time.

I think I will close this issue and publish v11.
I will open a separate issue for the test.

**Comment by s4m0r4m4** [comment]
Sounds good!
